types/funds_test.go

```diff
@@ -0,0 +1,200 @@
+package types
+
+import (
+	"math/big"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+var testData map[string]Funds = map[string]Funds{
+	"blank": {},
+	"zeros": {
+		common.HexToAddress("0x00"): big.NewInt(0),
+		common.HexToAddress("0x01"): big.NewInt(0),
+		common.HexToAddress("0x02"): big.NewInt(0),
+		common.HexToAddress("0x03"): big.NewInt(0),
+		common.HexToAddress("0x0a"): big.NewInt(0),
+		common.HexToAddress("0x0b"): big.NewInt(0),
+		common.HexToAddress("0x0c"): big.NewInt(0),
+	},
+	"a": {
+		common.HexToAddress("0x00"): big.NewInt(1),
+	},
+	"b": {
+		common.HexToAddress("0x00"): big.NewInt(1),
+	},
+	"c": {
+		common.HexToAddress("0x01"): big.NewInt(1),
+	},
+	"d": {
+		common.HexToAddress("0x02"): big.NewInt(1),
+	},
+	"e": {
+		common.HexToAddress("0x00"):  big.NewInt(1),
+		common.HexToAddress("0xabc"): big.NewInt(0),
+	},
+	"f": {
+		common.HexToAddress("0x00"): big.NewInt(2),
+	},
+	// manually calculated sums
+	"ab": {
```

Comment:
Nit: could you key this as "a+b" (and so on for the other manually calculated entries)?